lib/Segments/DynamicSegments/Filters/EmailActionClickAny.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace MailPoet\Segments\DynamicSegments\Filters;
+
+use MailPoet\Entities\DynamicSegmentFilterEntity;
+use MailPoet\Entities\NewsletterLinkEntity;
+use MailPoet\Entities\StatisticsClickEntity;
+use MailPoet\Entities\SubscriberEntity;
+use MailPoetVendor\Doctrine\DBAL\Query\QueryBuilder;
+use MailPoetVendor\Doctrine\ORM\EntityManager;
+
+class EmailActionClickAny implements Filter {
```

Comment:
Is extraction of `EmailActionClickAny` to a new class related to refactoring of the "clicked" segment? Is there any specific reason why it's done in this ticket and not a follow-up one?

Reply:
The reason was to make my job easier. There were so many options in the EmailAction segment that it was hard to update it. I had to add some more `if` branches and the code would become harder to read. I decided to refactor this filter to a separate file. It has nothing to do with the other filters anyway.